src/physics/base/ParticleInterface.hh

```diff
@@ -86,19 +108,47 @@ struct ParticleTrackState
  * The size of the view will be the size of the vector of tracks. Each particle
  * track state corresponds to the thread ID (\c ThreadId).
  *
- * \sa ParticleStateStore (owns the pointed-to data)
  * \sa ParticleTrackView (uses the pointed-to data in a kernel)
  */
-struct ParticleStatePointers
+template<Ownership W, MemSpace M>
+struct ParticleStateData
 {
-    Span<ParticleTrackState> vars;
+    template<class T>
+    using Data = celeritas::StatePie<T, W, M>;
+
+    Data<ParticleTrackState> state;
 
-    //! Check whether the interface is initialized
-    explicit CELER_FUNCTION operator bool() const { return !vars.empty(); }
+    //! Whether the interface is assigned
+    explicit CELER_FUNCTION operator bool() const { return !state.empty(); }
 
     //! State size
-    CELER_FUNCTION size_type size() const { return vars.size(); }
+    CELER_FUNCTION size_type size() const { return state.size(); }
+
+    //! Assign from another set of data
+    template<Ownership W2, MemSpace M2>
+    ParticleStateData& operator=(ParticleStateData<W2, M2>& other)
+    {
+        CELER_EXPECT(other);
+        state = other.state;
+        return *this;
+    }
 };
 
+#ifndef __CUDA_ARCH__
+//---------------------------------------------------------------------------//
+/*!
+ * Resize particle states in host code.
+ */
+template<MemSpace M>
+inline void
+resize(ParticleStateData<Ownership::value, M>* data,
+       const ParticleParamsData<Ownership::const_reference, MemSpace::host>&,
```

Comment:
Unused? Will we need this in the future?

Reply:
Ah, this unused parameter is for consistency with the material resizer (and upcoming physics resizer). Conceivably the state initialization could require some runtime parameter data besides the number of tracks...